plugins/browser-plugin-performance-navigation-timing/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { BrowserPlugin } from '@snowplow/browser-tracker-core';
+import { constructNavigationTimingContext, getPerformanceNavigationTimingContext } from './contexts';
+import { SelfDescribingJson } from '@snowplow/tracker-core';
+
+let navigationTimingEntry: SelfDescribingJson<Record<string, unknown>>[];
+
+/**
+ * Adds Performance Navigation Timing context to events
+ *
+ * @remarks
+ * May not be fully populated when initIal Page View fires
+ * Often a good idea to take the latest performance timing context
+ * for a given page view id when analyzing in the warehouse
+ */
+export function PerformanceNavigationTimingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: () => {
+      /* Setup a PerformanceObserver to allow caching of a completed navigation timing entry. We don't need to query the API after that. */
+      if (typeof PerformanceObserver !== 'undefined') {
+        const perfObserver = new PerformanceObserver((observedEntries) => {
+          const [observedNavigationTiming] = observedEntries.getEntries() as PerformanceNavigationTiming[];
+          /* We mark the entry as final by the measurement of loadEventEnd which is the final in the processing model  */
+          if (observedNavigationTiming.loadEventEnd > 0) {
+            navigationTimingEntry = constructNavigationTimingContext(observedNavigationTiming);
```

Comment:
To be honest, I was thinking that it would be fine to keep the observer still connected for the navigation type.

After this point, we do not expect it to be triggered again and the memory footprint of leaving it connected will be negligible afaik. I personally think that leaving it running might allow the client to detect issues that might have to do with browser navigation moving forwards _(more events than expected or some bad events)_.

I might be nitpicking here, wdyt ?

Reply:
Yeah fair enough, I doubt the performance impact will be significant so the leak is fine. And yeah if it helps us notice some issue down the line it's probably worth it.